src/lib/stringviews.nim

```diff
@@ -43,17 +46,27 @@ proc add*(s: var string; b: StringView) =
   for i in 0..<b.len:
     s[l+i] = b[i]
 
+when not defined(nimony):
+  proc rawData(s: string): ptr UncheckedArray[char] {.inline.} =
+    if s.len == 0:
+      nil
+    else:
+      cast[ptr UncheckedArray[char]](addr(s[0]))
+
 proc `$`*(s: StringView): string =
   result = newString(s.len)
   if s.len > 0:
-    copyMem addr(result[0]), s.p, s.len
+    copyMem result.rawData, s.p, s.len
 
 proc toStringViewUnsafe*(s: string): StringView =
   ## Watch out that the string lives longer than the string view!
   if s.len != 0:
-    result = StringView(p: cast[pchar](addr(s[0])), len: s.len)
+    result = StringView(p: cast[pchar](s.rawData), len: s.len)
```

Comment:
`addr(s[0])` seems to work if `[]` in `stringimpl` is changed to have a return type of `var char`

Reply:
Nope. Can't do that with the new strings.